baybe/recommenders/deprecation.py

```diff
@@ -36,6 +37,6 @@ def structure_recommender_protocol(val: dict, _) -> RecommenderProtocol:
             f"a future version.",
             DeprecationWarning,
         )
-    fun = make_dict_structure_fn(cls, converter)  # type: ignore
+    fun = make_dict_structure_fn(cls, converter, _cattrs_forbid_extra_keys=True)  # type: ignore
```

Comment:
This does not seem to be related to `objectives`, could you explain the changes made in this file?

Reply:
fyi Adrian I think the remaining issues referenced [here](https://github.com/emdgroup/baybe/issues/156#issuecomment-2020663266) were related to the allowing of additional keywords

Reply:
Yes, related (see explanation in other thread) but I think it doesn't solve the misspelling issue entirely – though it's a good first step 👍🏼 Ideally, I would like to disable the keywords not on the hook level but directly for the entire converter itself, which is currently not yet possible due to my custom-build tagged union strategy (i.e. the hook mechanism that allows us to deserialize arbitrary subclasses of a class). As a reminder: I implemented this before `cattrs` offered a built-in solution for it. Now they have it, and I hope that we can switch soon, but some of the existing code still blocks us from doing so.

Reply:
will resolve (has been discussed in other thread already)